sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -60,6 +60,37 @@ static const bool UseCopyEngineForD2DCopy = [] {
   return (CopyEngineForD2DCopy && (std::stoi(CopyEngineForD2DCopy) != 0));
 }();
 
+// SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE can be set to an integer value, or
+// a pair of integer values of the form "lower_index:upper_index".
+// Here, the indices point to copy engines in a list of all available copy
+// engines.
+// This functions returns this pair of indices.
+// If the user specifies only a single integer, a value of 0 indicates that
+// the copy engines will not be used at all. A value of 1 indicates that all
+// available copy engines can be used.
+static const std::pair<int, int> getRangeOfAllowedCopyEngines = [] {
+  std::string CopyEngineRange =
+      std::getenv("SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE");
+  // If the environment variable is not set, all available copy engines can be
+  // used.
+  if (CopyEngineRange.empty()) {
+    return std::pair<int, int>(0, INT_MAX);
+  }
+  // Environment variable can be a single integer or a pair of integers
+  // separated by ":"
+  auto pos = CopyEngineRange.find(":");
+  if (pos == std::string::npos) {
+    bool UseCopyEngine = (std::stoi(CopyEngineRange) != 0);
+    if (UseCopyEngine)
+      return std::pair<int, int>(0, INT_MAX); // All copy engines can be used.
+    else
+      return std::pair<int, int>(-1, -1); // No copy engines will be used.
+  }
+  int LowerCopyEngineIndex = std::stoi(CopyEngineRange.substr(0, pos));
+  int UpperCopyEngineIndex = std::stoi(CopyEngineRange.substr(pos + 1));
+  return std::pair<int, int>(LowerCopyEngineIndex, UpperCopyEngineIndex);
+}();
+
 static const bool CopyEngineRequested = [] {
```

Comment:
please remove this and use the new "range" function instead?

Reply:
It looks a bit cleaner this way. If we piggyback on the 'range' function, we will have to add additional logic at the caller site to check the value of the pair returned by the 'range' function. The good thing I like about this function is that it handles only one use case: SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE=0; for any other case, it will return true.

Reply:
But now you have to remember to fix 2 functions if smth needs to be changed. Can you at least implement this function via the new one?

Reply:
Ok..done...thanks